payjoin/tests/integration.rs

```diff
@@ -79,6 +132,45 @@ mod integration {
             );
             Ok(())
         }
+
+        #[test]
+        fn disallow_mixed_input_scripts() -> Result<(), BoxError> {
+            init_tracing();
+            let (_bitcoind, sender, receiver) = init_bitcoind_sender_receiver(
+                Some(AddressType::Bech32),
+                Some(AddressType::P2shSegwit),
+            )?;
+
+            // Receiver creates the payjoin URI
+            let pj_receiver_address = receiver.get_new_address(None, None)?.assume_checked();
+            let pj_uri = PjUriBuilder::new(pj_receiver_address, EXAMPLE_URL.to_owned())
+                .amount(Amount::ONE_BTC)
+                .build();
+
+            // **********************
+            // Inside the Sender:
+            // Sender create a funded PSBT (not broadcasted) to address with amount given in the pj_uri
+            let uri = Uri::from_str(&pj_uri.to_string())
+                .unwrap()
+                .assume_checked()
+                .check_pj_supported()
+                .unwrap();
+            let psbt = build_original_psbt(&sender, &uri)?;
+            debug!("Original psbt: {:#?}", psbt);
+            let (req, ctx) = RequestBuilder::from_psbt_and_uri(psbt, uri)?
+                .build_with_additional_fee(Amount::from_sat(10000), None, FeeRate::ZERO, false)?
+                .extract_v1()?;
+            let headers = HeaderMock::new(&req.body, req.content_type);
+
+            // **********************
+            // Inside the Receiver:
+            // This should error because the receiver is attempting to introduce mixed input script types
+            let result = std::panic::catch_unwind(std::panic::AssertUnwindSafe(|| {
+                handle_v1_pj_request(req, headers, &receiver, None, None, None)
+            }));
+            assert!(result.is_err());
```

Comment:
`handle_v1_pj_request` unwraps all errors and it seemed non-trivial to make the actual error bubble up instead. This is a workaround to catch the panic and ensure _some_ error occurred.

Reply:
All the panics in the integration test are annoying, but the fix is probably simpler than you thought. Most of those errors can be boxed and bubbled up without a custom error type as `Result<_, Box<dyn std::error::Error + 'static>` and the ones that can't can be addressed with `map_err`.